### PR TITLE
issue_717: NRFU6.10: Update skip enhancement nrfu_tests/test_hardware_power_supply_voltage_status.py

### PR DESCRIPTION
**Pull request by rahulmali2512**
# Please include a summary of the changes

* Updated test_hardware_power_supply_voltage_status.py testcase with skip enhancement (NRFU6.10)

# Any specific logic/part of code you need extra attention on

Explain any specific logic you need special attention on

# Include the Issue number and link
https://github.com/aristanetworks/vane/issues/717

# List/Attach any dependencies/past issues that are required for this change/provide context

# Type of change

- - [ ] Bug fix (non-breaking change which fixes an issue)
- - [ ] New feature (non-breaking change which adds functionality)
- - [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- - [ ] This change requires a documentation update
- - [x] Other (please specify)

# Effort required on reviewers end

- - [x] Easy
- - [ ] Medium
- - [ ] Hard 

# How Has This Been Tested?

## Bug fix

    Include before and after snapshots/screenshots/changed logs
    
## New feature

    Ensure current test cases pass and include newer test cases if required
    
# CI pipeline result

- - [ ] Pass
- - [ ] Fail
  
  If it fails, explain the reason and whether or not we should ignore the failure
  
# Verify Documentation Update

    If applicable, ensure documentation has been updated for ReadMe, Getting Started guide, and Style guide
    
# Additional comments

- Skipped reports when device is vEOS device.

[report_skip.docx](https://github.com/aristanetworks/vane/files/14721232/report_skip.docx)
[report_pass.docx](https://github.com/aristanetworks/vane/files/14721233/report_pass.docx)

- HTML reports

[html reports.zip](https://github.com/aristanetworks/vane/files/14721237/html.reports.zip)

### CODE REVIEW

File: nrfu_tests/test_hardware_power_supply_voltage_status.py [hunk 1]
```diff
@@ -114,7 +120,12 @@ def test_system_hardware_power_supply_voltage_status(self, dut, tests_definition
                         f" follows:\n{power_supply_status}"
                     )
 
-        except (AssertionError, AttributeError, LookupError, EapiError) as excep:
+        # For BaseException test case is failing instead of skipping it. Hence, adding
+        # specific exception here.
+        except pytest.skip.Exception:
+            pytest.skip(tops.output_msg)
+
+        except (BaseException, EapiError) as excep:
```
Comment (dhruvgoel9) [inline nrfu_tests/test_hardware_power_supply_voltage_status.py:128]: Revert the changes of except block, don't add pytest skip exception

File: nrfu_tests/test_hardware_power_supply_voltage_status.py [hunk 2]
```diff
@@ -57,33 +57,39 @@ def test_system_hardware_power_supply_voltage_status(self, dut, tests_definition
...
                 if "PowerSupply" in sensor
             ]
             assert (
-                power_voltage_sensor
+                self.power_voltage_sensor
             ), "Power supply voltage sensor details are not found on the device."
 
             # Updating the actual and expected state of the power supply sensor
-            for voltage_sensor in power_voltage_sensor:
+            for voltage_sensor in self.power_voltage_sensor:
```
Comment (dhruvgoel9) [inline nrfu_tests/test_hardware_power_supply_voltage_status.py:92]: Why these changes are required

**Review by dhruvgoel9 — COMMENTED** [review]
Update the task accordingly

File: nrfu_tests/test_hardware_power_supply_voltage_status.py [hunk 3]
```diff
@@ -69,7 +73,9 @@ def test_system_hardware_power_supply_voltage_status(self, dut, tests_definition
                 f"On device {tops.dut_name}, output of {power_supply_cmd} command is:"
                 f" \n{voltage_cmd_output}\n"
             )
-            self.output += f"\nOutput of {power_supply_cmd} command is:\n{voltage_cmd_output}\n"
+            self.output += (
+                f"\nOutput of {power_supply_cmd} command is:\n{voltage_cmd_output}\n"
```
Comment (dhruvgoel9) [inline nrfu_tests/test_hardware_power_supply_voltage_status.py:77]: Why we have made these changes

**Review by dhruvgoel9 — APPROVED** [review]

**Review by geetanjalimanegslab — APPROVED** [review]

File: nrfu_tests/test_hardware_power_supply_voltage_status.py [hunk 4]
```diff
@@ -58,7 +58,11 @@ def test_system_hardware_power_supply_voltage_status(self, dut, tests_definition
             # Skipping test case if the device is vEOS.
             model = version_output.get("modelName")
             if "vEOS" in version_output.get("modelName"):
```
Comment (rewati-arista) [inline nrfu_tests/test_hardware_power_supply_voltage_status.py:60]: Why are we not using model here?
```suggestion
            if "vEOS" in model:
```

**Review by rewati-arista — CHANGES_REQUESTED** [review]
One change requested about variable usage

**Review by rewati-arista — APPROVED** [review]

**Pull request merged**